staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -413,10 +413,15 @@ type DeploymentSpec struct {
 }
 
 const (
+	// Unprefixed labels are reserved for end-users
+	// so we will add a kubernetes.io to designate these labels as official Kubernetes labels.
+	labelPrefix = "kubernetes.io/"
+	// LegacyDeploymentUniqueLabelKey is a legacy label that was set using unprefixed labels.
+	LegacyDeploymentUniqueLabelKey = "pod-template-hash"
 	// DefaultDeploymentUniqueLabelKey is the default key of the selector that is added
 	// to existing ReplicaSets (and label key that is added to its pods) to prevent the existing ReplicaSets
 	// to select new pods (and old pods being select by new ReplicaSet).
-	DefaultDeploymentUniqueLabelKey string = "pod-template-hash"
+	DefaultDeploymentUniqueLabelKey string = labelPrefix + "pod-template-hash"
```

Comment:
I'm not sure if changing the value is ok.
Let's say a developer upgrades the k8s.io/api library, but a user of a controller runs in an older kubernetes version. That component would break.

A better approach would be to keep this variable as is, mark it as Deprecated, and add a new variable called `FullyQualifiedDeploymentUniqueLabelKey` or just `DeploymentUniqueLabelKey` (not sure why we had Default on it before).

Reply:
I like the idea of not even having `DefaultDeploymentUniqueLabelKey`, if our code-level API guarantees would allow that.